Lib/test/test_datetime.py

```diff
@@ -20,7 +20,7 @@
 # XXX(gb) First run all the _Pure tests, then all the _Fast tests.  You might
 # not believe this, but in spite of all the sys.modules trickery running a _Pure
 # test last will leave a mix of pure and native datetime stuff lying around.
-test_classes = []
+all_test_classes = []
```

Comment:
Because the tests were only running for the C implementation, not for the Python 
implementation. This was because `test_classes` was being overwritten inside the `for` loop.  With this change, the tests run twice: once for the pure Python module and once for the C extension.